prompting/tasks/benchmark.py

```diff
@@ -0,0 +1,150 @@
+            context=context.content, title=context.title, source=context.source
+        )
+
+        query_with_choices = self.generate_query(llm_pipeline)
+
+        self.query, self.reference = self.extract_query_and_reference(
+            query_with_choices
+        )
+
+        self.topic = context.title
+        self.subtopic = context.topic
+        self.tags = context.tags
+
+    def extract_query_and_reference(self, query_with_choices: str) -> Tuple[str, str]:
+        """
+        Extract the query and reference answer from the generated query.
+        To do this we extract the reference answer by searching for the choice with a * symbol,
+        and then removing the * to form the query
+        """
+        # get the index of first occurrence of the choices
+        index = re.search(self.choices_pattern, query_with_choices).start()
+
+        items, choices = list(
+            zip(*re.findall(self.choices_pattern, query_with_choices[index:]))
+        )
+        if len(choices) != 4:
+            raise TaskCreationError(
+                f"{self.__class__.__name__} the number of choices is not 4 in query_with_choices:\n{query_with_choices}"
+            )
+
+        correct_item = [i for i, item in enumerate(items) if "*" in item]
+        if len(correct_item) == 0:
+            raise TaskCreationError(
+                f"{self.__class__.__name__} no reference was found in query_with_choices:\n{query_with_choices}"
+            )
+        elif len(correct_item) != 1:
+            raise TaskCreationError(
+                f"{self.__class__.__name__} found multiple reference matches in query_with_choices:\n{query_with_choices}"
+            )
+        reference_label = choices[correct_item[0]]
+
+        shuffled_choices, new_reference = self.shuffle_choices(choices, reference_label)
+        shuffled_query_with_choices = (
+            query_with_choices[:index] + "\n\n" + shuffled_choices
+        )
+        return shuffled_query_with_choices, new_reference
+
+    def shuffle_choices(
+        self, choices: List[str], reference_label: str
+    ) -> Tuple[str, str]:
+        """Shuffle the choices and return the new reference. By itself, the LLM will almost always assign the reference to option B or C.
+        This method overcomes the highly biased ordering by manually reshuffling the choices so that the reference is uniformly distributed across the options.
+
+        Args:
+            choices (list): list of choices
+            reference_label (str): the reference answer label
+
+        Returns:
+            str: the shuffled choices
+            str: the new reference, in {A, B, C, D}
+        """
+        # shuffle the choices
+        shuffled_choice_list = list(
+            zip("ABCD", np.random.choice(choices, len(choices), replace=False))
+        )
+
+        # match the reference and get the letter
+        new_reference = [c[0] for c in shuffled_choice_list if c[1] == reference_label][
+            0
+        ]
```

Comment:
It loops through the list of all choices, even if reference was already found, minor optimization:
```suggestion
        new_reference = next(label for label, choice in shuffled_choice_list if choice == reference_label)
```